From 54c282cb8ac61da9ede4f05b86dbe59116f71e69 Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Date: Wed, 4 Jan 2012 21:18:12 +0200
Subject: egl: test: Check that eglGetConfigs checks for NULL parameter

Passing NULL as the last paramter of eglGetConfigs is required to
generate EGL_BAD_PARAMETER. Add unit test to check that eglGetConfigs
functions correctly.

Change-Id: I22fb382ca84334237eafbe291e3f943d2045944d
Original-Change-Id: I9c998873c08fe60b2bcf6ae73da5954c79da3dda
Signed-off-by: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Signed-off-by: Russell Webb <russell.webb@intel.com>
---
 opengl/tests/EGLTest/EGL_test.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/opengl/tests/EGLTest/EGL_test.cpp b/opengl/tests/EGLTest/EGL_test.cpp
index c0daba2..67c1d0f 100644
--- a/opengl/tests/EGLTest/EGL_test.cpp
+++ b/opengl/tests/EGLTest/EGL_test.cpp
@@ -54,6 +54,13 @@ protected:
     }
 };
 
+TEST_F(EGLTest, EGLGetConfigsWithNullNumConfigs) {
+    EGLBoolean success;
+    success = eglGetConfigs(mEglDisplay, NULL, 0, NULL);
+    ASSERT_EQ(EGL_FALSE, success);
+    ASSERT_EQ(EGL_BAD_PARAMETER, eglGetError());
+}
+
 TEST_F(EGLTest, DISABLED_EGLConfigEightBitFirst) {
 
     EGLint numConfigs;
-- 
1.7.4.1

